rollup.config.mjs

```diff
@@ -0,0 +1,172 @@
+// @ts-check
+
+import fs from 'node:fs';
+import { createRequire } from 'node:module';
+import path from 'node:path';
+import terserPlugin from '@rollup/plugin-terser';
+import tsPlugin from '@rollup/plugin-typescript';
+import dtsPlugin from 'rollup-plugin-dts';
+
+/**
+ * @type {{
+ *   exports: Record<string, string>;
+ *   publishConfig: { browser: string };
+ * }}
+ */
+const packageJson = createRequire(import.meta.url)('./package.json');
+
+const testPatterns = ['**/*.bench.ts', '**/*.spec.ts', '**/*.test.ts'];
+
+export default () => {
+  clearDir('dist');
+  clearDir('umd');
+
+  const entrypoints = Object.values(packageJson.exports).filter(f => /^(\.\/)?src\//.test(f) && f.endsWith('.ts'));
```

Comment:
After a bunch of testing: using the paths from `package.json#exports` (the dev version pointing to the `src` directory) as inputs to Rollup produces the best results:

- The whole module tree gets resolved correctly.
- This avoids pulling in test files, and modules not marked as test files but only consumed by tests (e.g. `src/compat/_internal/*.ts`).

Another approach is to use something like `globSync('src/**/*.ts')` and then do a lot of filtering, but I found that it's easy to forget something while filtering.